src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (const action of actions) {
+    const { type } = action;
+    const currentState = { ...resultArray[resultArray.length - 1] || state };
+
+    if (type === 'addProperties') {
+      const { extraData } = action;
+
+      Object.assign(currentState, extraData);
+    } else if (type === 'removeProperties') {
+      const { keysToRemove } = action;
+
+      for (const key of keysToRemove) {
+        if (currentState.hasOwnProperty(key)) {
```

Comment:
The same as the previous comment. The 'delete' operator can be used without the 'hasOwnProperty' check.